flusher.go

```diff
@@ -19,107 +20,70 @@ import (
 	"github.com/stripe/veneur/trace"
 )
 
-const DatadogResourceKey = "resource"
-
 // Flush collects sampler's metrics and passes them to sinks.
-func (s *Server) Flush() {
+func (s *Server) Flush(ctx context.Context) {
 	span := tracer.StartSpan("flush").(*trace.Span)
 	defer span.Finish()
 
+	// TODO Move this to an independent ticker routine or something?
```

Comment:
So, I don't think we can do this in a separate goroutine, because we want this to be reporting the memory at the time of maximum usage (right before we clear and flush everything).


The alternate way of looking at it: we *need* to do it in a separate ticker, and choose our interval for that ticker such that we are sampling the underlying, sinusoidal data accurately.


Either way, this would also be a great place to report softnet data, once this is merged: https://github.com/stripe/veneur/pull/291

Reply:
This is separately addressed in #294